pkg/apis/build/v1beta1/build_conversion.go

```diff
@@ -0,0 +1,381 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package v1beta1
+
+import (
+	"context"
+
+	"github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/shipwright-io/build/pkg/ctxlog"
+	"github.com/shipwright-io/build/pkg/webhook"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	betaGroupVersion  = "shipwright.io/v1beta1"
+	alphaGroupVersion = "shipwright.io/v1alpha1"
+)
+
+// ensure v1beta1 implements the Conversion interface
+var _ webhook.Conversion = (*Build)(nil)
+
+func (src *Build) ConvertTo(ctx context.Context, obj *unstructured.Unstructured) error {
+	var bs v1alpha1.Build
+
+	bs.TypeMeta = src.TypeMeta
+	bs.TypeMeta.APIVersion = alphaGroupVersion
+
+	bs.ObjectMeta = src.ObjectMeta
+
+	src.Spec.ConvertTo(&bs.Spec)
+	mapito, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&bs)
+	if err != nil {
+		ctxlog.Error(ctx, err, "failed structuring the newObject")
+	}
+	obj.Object = mapito
+
+	return nil
+
+}
+
+func (src *Build) ConvertFrom(ctx context.Context, obj *unstructured.Unstructured) error {
+
+	var bs v1alpha1.Build
+
+	unstructured := obj.UnstructuredContent()
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructured, &bs)
+	if err != nil {
+		ctxlog.Error(ctx, err, "failed unstructuring the convertedObject")
+	}
+	src.ObjectMeta = bs.ObjectMeta
+	src.TypeMeta = bs.TypeMeta
+	src.TypeMeta.APIVersion = betaGroupVersion
+
+	src.Spec.ConvertFrom(&bs.Spec)
+
+	src.Status = BuildStatus{
+		Registered: bs.Status.Registered,
+		Reason:     (*BuildReason)(bs.Status.Reason),
+		Message:    bs.Status.Message,
+	}
+
+	return nil
+}
+
+func (dest *BuildSpec) ConvertFrom(orig *v1alpha1.BuildSpec) error {
+	// Handle BuildSpec Source
+	specSource := Source{}
+	if orig.Source.BundleContainer != nil {
+		specSource.Type = OCIArtifactType
+		specSource.OCIArtifact = &OCIArtifact{
+			Image:      orig.Source.BundleContainer.Image,
+			Prune:      (*PruneOption)(orig.Source.BundleContainer.Prune),
```

Comment:
will not, not to sure why, but somehow go handles this gracefully, giving back a nil. It might be that we create a pointer to Prune(which is nil) therefore we get a pointer with nil.